rpc-rs/src/provider.rs

```diff
@@ -521,34 +527,43 @@ impl HostBridge {
             .subscribe(shutdown_topic)
             .await
             .map_err(|e| RpcError::Nats(e.to_string()))?;
-        let msg = sub.next().await;
-        // TODO: there should be validation on this message, but it's not signed by host yet
-        // Shutdown messages are unsigned (see https://github.com/wasmCloud/wasmcloud-otp/issues/256)
-        // so we can't verify that this came from a trusted source.
-        // When the above issue is fixed, verify the source and keep looping if it's invalid.
-
-        info!("Received termination signal. Shutting down capability provider.");
-        // Tell provider to shutdown - before we shut down nats subscriptions,
-        // in case it needs to do any message passing during shutdown
-        if let Err(e) = provider.shutdown().await {
-            error!(error = %e, "got error during provider shutdown processing");
-        }
-        // send ack to host
-        if let Some(async_nats::Message { reply: Some(reply_to), .. }) = msg {
-            let data = b"shutting down".to_vec();
-            if let Err(e) = self.rpc_client().publish(reply_to, data).await {
-                error!(error = %e, "failed to send shutdown ack");
-            }
-        }
 
-        // unsubscribe from shutdown topic
-        let _ = sub.unsubscribe().await;
+        loop {
+            let msg = sub.next().await;
+            // TODO: there should be validation on this message, but it's not signed by host yet
+            // Shutdown messages are unsigned (see https://github.com/wasmCloud/wasmcloud-otp/issues/256)
+            // so we can't verify that this came from a trusted source.
+            // When the above issue is fixed, verify the source and keep looping if it's invalid.
+
+            info!("Received termination signal.");
+
+            // Check if we really need to shut down
+            if let Some(async_nats::Message { reply: Some(reply_to), payload, .. }) = msg {
+                let shutmsg: ShutdownMessage = serde_json::from_slice(&payload).unwrap_or_default();
+                if shutmsg.host_id == self.host_data.host_id {
```

Comment:
As unfortunate as it is, I think this change necessitates a breaking change in wasmbus-rpc

Scenario:
1. Run wasmCloud version `0.58.2` which isn't patched 
2. Run a provider with this version of wasmbus-rpc
3. If the older versions of host send a shutdown signal, after this version the shutdown signal is completely ignored since it didn't include the host ID before

Reply:
valid point.